Update plugin.py to add support for mkdocs-autorefs

stephffuller opened the pull request:
@timvink Cavaet - I'm no developer, and this was written with Claude Sonnet 4 over several trial-and-error iterations. I tested in my output pdf, and the autoref links are working as expected. YMMV.

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+        # As this plugin processes cross-references in the on_env event, 
+        # which happens after the print page is generated, it's necessary to 
+        # manually execute the autorefs fix_refs function here.
+        logger.info(f"[mkdocs-print-site] Available plugins: {list(config.get('plugins', {}).keys())}")
```
Please remove this logging line

stephffuller replied:
Done.

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+            )
+            if unmapped:
+                logger.warning(f"[mkdocs-print-site] Unmapped autorefs: {[ref for ref, _ in unmapped]}")
+        else:
+            logger.info("[mkdocs-print-site] No autorefs plugin found")   
```
You can remove this also

stephffuller replied:
Done

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+        logger.info(f"[mkdocs-print-site] Available plugins: {list(config.get('plugins', {}).keys())}")
+        autorefs_plugin = config.get("plugins", {}).get("mkdocs-autorefs") or config.get("plugins", {}).get("autorefs")
+        if autorefs_plugin:
+            logger.info("[mkdocs-print-site] Processing autorefs cross-references...")
```
No logging please. If you really need it, use `logger.debug` instead

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+        autorefs_plugin = config.get("plugins", {}).get("mkdocs-autorefs") or config.get("plugins", {}).get("autorefs")
+        if autorefs_plugin:
+            logger.info("[mkdocs-print-site] Processing autorefs cross-references...")
+            import functools
```
This can be a toplevel import

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+            from mkdocs_autorefs._internal.references import fix_refs
+            
+            # First, extract all available anchors from the HTML
+            import re as regex_module
```
top level import also

timvink commented on src/mkdocs_print_site_plugin/plugin.py:
```diff
@@ -330,6 +330,141 @@ def on_post_build(self, config, **kwargs):
+        else:
+            logger.info("[mkdocs-print-site] No autorefs plugin found")   
+
+        # Alternative autorefs implementation when plugin isn't available
+        if not autorefs_plugin:
```
Here we are rebuilding autorefs functionality. Let's remove this entire section, and just use the implementation for when autorefs plugin is activated.

stephffuller replied:
Removed.

timvink reviewed (changes requested):
Great job, by both you and claude :)

Left some comments, but it looks good.

We'll also need to update the `README.md` and specify which plugins are supported, and include autorefs. Can you do that also?

stephffuller commented:
Ok @timvink , I think I've addressed all your comments. I also tested and the autorefs are still working in the html preview site and in the generated pdf after these latest changes. 

The README update ended up in a different PR for some reason. I think I had too many GH tabs open and committed in the wrong one or something. This one is ready for re-review.

timvink commented:
Back from holidays :) 

Thanks for the changes!

timvink commented:
Released
